Src/World/ScriptWorld.cpp

```diff
@@ -141,8 +161,13 @@ ScriptWorld::~ScriptWorld() {
     }
 
     delete perTickEventDefinition;
+    delete perEveryTickEventDefinition;
+    delete perLoadTickEventDefinition;
     delete perFrameGameEventDefinition;
-    delete perFrameMenuEventDefinition;
+    delete perMenuFrameEventDefinition;
+    delete perLoadFrameEventDefinition;
+    delete perEveryFrameEventDefinition;
+    delete resolutionChangedEventDefinition;
```

Comment:
A clearer explanation of these events is required.

Reply:
Load Tick, and Load Frame, are a state that can be triggered by angelscript that causes it to only run those hooks.
It felt like a cleaner implementation for modding purposes, saves every mod doing if(Environment::Loading).
This should probably be further iterated into giving each mod its own loading states and cycles, particularly to account for dependencies, load order, and mods modifying the outputs of other mods.
Main reason to have this now is to build as though this state is separate from the primary tick and tickAlways functions, and particularly to reduce and/or manage initial load times in the immediate term, making it faster to test code iterations and changes.

For the other events;

Ticking;
updateTick happens immediately after tick++ happens. tick++ only happens if, and only if the game is not paused.
updateEveryTick happens immediately before the next tick++ happens.
It is set to trigger in this particular fashion (World.cpp:176) to allow advanceOneTick() to be created, such that the first EveryTick() unpauses the game, and marks the next EveryTick() to pause it again.

Framing;
updateFrame is intended for frame advancement if, and only if the game is not paused, such as advancing animation sequences.
updateEveryFrame is intended for rendering things that should always be rendered, like the world, objects, and npcs in view.
I understand that framing is going to be done differently from what I have been thinking, and I have a suspicion that only updateFrame may be needed, where the engine may retain the world state and does not need additional render calls to continue rendering things in their current state.